src/artm/core/dictionary.h

```diff
@@ -46,7 +46,7 @@ class Dictionary {
   std::vector<DictionaryEntry> entries_;
   std::unordered_map<Token, int, TokenHasher> token_index_;
   std::unordered_map<int, Token> index_token_;
-  std::unordered_map<int, std::unordered_map<int, int> > cooc_values_;
+  std::unordered_map<int, std::unordered_map<int, std::vector<float> > > cooc_values_;
```

Comment:
I'm a bit concerned about storing so many std::vector<float>, particularly when each of them has only 1 value. Look at sizeof(std::vector<float>), which is 24 bytes on 64 bit. Plus the actual memory for 1 float. In total this is 7 times more memory usage than you really need. 

Again, I'm not convinced that we need to support multiple columns in cooc file - this makes the API more complicated, plus user needs to remember which columns is for what purpose. To me it is far more simple to create several dictionaries in BigARTM.  What do you think? 
If you really want to support multiple columns in cooc file then I would suggest to store them as follows:
std::vector<std::unordered_map<int, std::unordered_map<int, int> >>  cooc_values_.
I think this will use less memory (even though it duplicates all keys).
This option will only work if all of your std::vector<float> had the same length, but I think this is required anyway...